dashboard/src/components/Visualization.js

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect, useState, useRef } from 'react';
+import createPlotlyComponent from 'react-plotly.js/factory'
+import Plotly from 'plotly.js';
+import { Col, Collapsible, CollapsibleItem, Icon } from 'react-materialize';
+
+
+const RENDERING_INTERVAL = 50;
+
+// CHART
+const CHART_PLOT_DURATION = 30 * 1000;
+const MINIMUM_DATA_AGE = 500;
+
+let timestampOffset = 0;
+
+const Plot = createPlotlyComponent(Plotly);
+
+const Visualization = (props) => {
+
+  // eslint-disable-next-line
+  const [dataRecordingContainerState, updateDataRecordingContainer] = useState(props.dataRecordingContainer);
+  const [layoutState, updateLayout] = useState({});
+  const [tracesState, updateTraces] = useState([{}]);
+  const recreateChartPlot = useRef(true);
+
+
+  useEffect(
+    () => {
+      const intervalID = window.setInterval(render, RENDERING_INTERVAL);
+      return () => clearInterval(intervalID);
+    },
+    // eslint-disable-next-line 
+    [props.selectedDataId]
```

Comment:
render should be a dependency of this effect.

Reply:
[props.selectedDataId, props.timestampOffset, render()]

Like this?

Reply:
not calling it, just the reference:
`[props.selectedDataId, timestampOffset, render, recreateChartPlot]`